java/ql/lib/semmle/code/java/Statement.qll

```diff
@@ -626,12 +616,26 @@ class BreakStmt extends Stmt, @breakstmt {
 /**
  * A `yield` statement.
  */
-class YieldStmt extends Stmt, @yieldstmt {
+class YieldStmt extends JumpStmt, @yieldstmt {
   /**
    * Gets the value of this `yield` statement.
    */
   Expr getValue() { result.getParent() = this }
 
+  /**
+   * Gets the `switch` expression target of this `yield` statement.
+   */
+  override SwitchExpr getTarget() {
+    // Get the innermost enclosing SwitchExpr; this works because getParent() is defined for Stmt and
+    // therefore won't proceed after the innermost SwitchExpr (due to it being an Expr)
+    result = this.getParent+()
+  }
+
+  /**
+   * DEPRECATED: `yield` statements cannot have a target label.
+   */
+  deprecated override LabeledStmt getTargetLabel() { none() }
```

Comment:
Have marked this as deprecated to prevent accidental usage. If that is a reasonable approach, it might be necessary to adjust [`config/blame-deprecations.mjs`](https://github.com/github/codeql/blob/main/config/blame-deprecations.mjs) (introduced by #8347) to not flag deprecated predicated which also have an `override` annotation. Otherwise this predicate override might be removed by accident in the future.

Reply:
Obsolete, see https://github.com/github/codeql/pull/8582#discussion_r837659375